File: src/main/java/net/hectus/neobb/turn/default_game/block/TPinkBed.java
package net.hectus.neobb.turn.default_game.block;

import com.marcpg.libpg.util.Randomizer;
import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.SupernaturalClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.block.Block;

import java.util.List;

public class TPinkBed extends BlockTurn implements BuffFunction, SupernaturalClazz {
    public TPinkBed(NeoPlayer player) { super(player); }
    public TPinkBed(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public void apply() {
        if (Randomizer.boolByChance(70)) {
            player.player.clearActivePotionEffects();
        }
    }

    @Override
    public List<Buff> buffs() {
        return List.of();
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/block/TPinkBed.kt
package net.hectus.neobb.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.Randomizer
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.SupernaturalClazz
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import org.bukkit.block.Block

class TPinkBed(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffFunction, SupernaturalClazz {
    override val cost: Int = 4

    override fun apply() {
        if (Randomizer.boolByChance(70.0)) {
            player!!.player.clearActivePotionEffects()
        }
    }

    override fun buffs(): List<Buff<*>> = emptyList()
}